### PR TITLE
Remove string-based indexers

### PR DESCRIPTION
**Pull request by jonsequitur**
This addresses #53 and partially addresses #1127.

### CODE REVIEW

File: src/System.CommandLine/Parsing/CommandResult.cs [hunk 1]
```diff
@@ -26,12 +24,8 @@ public class CommandResult : SymbolResult
...
         public ICommand Command { get; }
 
-        public OptionResult? this[string alias] => OptionResult(alias);
-
-        public OptionResult? OptionResult(string alias)
-        {
-            return Children[alias] as OptionResult;
-        }
+        // FIX: (CommandResult) remove indexer
+        public OptionResult? this[string alias] => Children.GetByAlias(alias) as OptionResult;
```
Comment (Keboo) [inline src/System.CommandLine/Parsing/CommandResult.cs:28]: The issue was remove or rationalize the indexers. Is there a reason we are leaving this one in?

File: src/System.CommandLine/Parsing/ParseResult.cs [hunk 2]
```diff
@@ -64,12 +64,20 @@ public class ParseResult
...
         public IReadOnlyCollection<string> UnparsedTokens { get; }
 
+        // FIX: (ParseResult) remove
         public object? ValueForOption(string alias) =>
             ValueForOption<object?>(alias);
         
+        public object? ValueForOption(Option option) =>
+            ValueForOption<object?>(option);
+
+        // FIX: (ParseResult) remove
```
Comment (Keboo) [inline src/System.CommandLine/Parsing/ParseResult.cs:74]: I am not sure what this comment means. Is there more to do here?

**Review by Keboo — APPROVED** [review]

File: src/System.CommandLine/Parsing/CommandResult.cs (hunk 1 above)
Comment (jonsequitur) [reply src/System.CommandLine/Parsing/CommandResult.cs]: I had forgotten to push the last commit.

File: src/System.CommandLine/Parsing/ParseResult.cs (hunk 2 above)
Comment (jonsequitur) [reply src/System.CommandLine/Parsing/ParseResult.cs]: Just a note to self. Removed.

**Pull request merged**